src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -2482,5 +2488,178 @@ describe('WorkspacesUsecases', () => {
         });
       });
     });
+
+    describe('deleteWorkspaceContent', () => {
+      it('When workspace is not found, then it should throw', async () => {
+        const workspaceId = v4();
+        const user = newUser();
+        jest.spyOn(workspaceRepository, 'findById').mockResolvedValue(null);
+
+        await expect(
+          service.deleteWorkspaceContent(workspaceId, user),
+        ).rejects.toThrow(NotFoundException);
+      });
+
+      it('When user is not the owner of the workspace, then it should throw', async () => {
+        const workspace = newWorkspace();
+        const user = newUser();
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+
+        await expect(
+          service.deleteWorkspaceContent(workspace.id, user),
+        ).rejects.toThrow(ForbiddenException);
+      });
+
+      it('When workspace is found and user is the owner, then it should delete all workspace content', async () => {
+        const user = newUser();
+        const workspace = newWorkspace({ attributes: { ownerId: user.uuid } });
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+
+        await service.deleteWorkspaceContent(workspace.id, user);
+
+        expect(workspaceRepository.deleteById).toHaveBeenCalledWith(
+          workspace.id,
+        );
+      });
+    });
+
+    describe('leaveWorkspace', () => {
+      it('When workspace is not found, then it should throw', async () => {
+        const workspaceId = v4();
+        const user = newUser();
+        jest.spyOn(workspaceRepository, 'findById').mockResolvedValue(null);
+
+        await expect(service.leaveWorkspace(workspaceId, user)).rejects.toThrow(
+          NotFoundException,
+        );
+      });
+
+      it('When user is not part of workspace, then it should throw', async () => {
+        const workspace = newWorkspace();
+        const user = newUser();
+        jest
+          .spyOn(workspaceRepository, 'findById')
+          .mockResolvedValue(workspace);
+        jest
+          .spyOn(workspaceRepository, 'findWorkspaceUser')
+          .mockResolvedValue(null);
+
+        await expect(
+          service.leaveWorkspace(workspace.id, user),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is the owner of the workspace, then deleteWorkspace should be called', async () => {
```

Comment:
We should only allow one way of removing the workspace. Is this the only way? If not, we should prevent the owner from leaving the workspace.

Reply:
I'm not sure, I would suppose there should be another way to specifically remove the workspace so this flow would be a bit confusing.  I'm now preventing the owner from leaving and left the deleteWorkspace usecase uncalled for the moment, as yes it should probably be executed from a separate endpoint.